### ISSUE TITLE
Initial client policies integration for SAML

### ISSUE THREAD

**Issue by rmartinc** [keycloak/keycloak#26654]
### Description

The addition of the client policies long ago in https://github.com/keycloak/keycloak/pull/7104 added the ability to enforce rules for clients. But they are more or less mainly thought for OIDC clients. Although some parts (like creation and update) are executed for SAML clients too, there are some gaps that need to be implemented for a full integration.

This issue is to implement the basics for SAML client policies and to implement some initial executors and conditions useful for them. The final idea is providing a client poliy profile similar to what we have for OIDC (fapi-1-baseline, fapi-2-security-profile,...).

In summary:

* Extend client policies to be executed in some SAML protocol points. For the moment two triggering points will be added (login and logout for saml), in which the client policies will be executed receiving the saml request, the client and more data that can be necessary. The policy conditions/exectors will be executed before performing the real operation.
* Add some initial executors and conditions similar to what we have for OIDC.
* Tests
* ...
